ksqldb-engine/src/main/java/io/confluent/ksql/planner/LogicalPlanner.java

```diff
@@ -627,6 +633,23 @@ private JoinNode buildJoin(final Join root, final String prefix, final boolean i
     );
   }
 
+  private Optional<ColumnName> getLeftJoinNonKeyColumn(final Join join) {
+    // This method is called after the verifyJoin() has detected a FK join. This code does not
+    // check for values are in the right order. It is assumed verifyJoin() already checked them
+    // and failed in case they are not valid.
+
+    // What if expression is not ColumnReferenceExp?
```

Comment:
We should throw for now, with a helpful error message. We can update this later when we want to support expressions (rather than limit FK joins to individual columns).

Reply:
Added the exception inside the `verifyJoin`.